storage/remote/storage.go

```diff
@@ -77,6 +77,9 @@ func NewStorage(l log.Logger, reg prometheus.Registerer, stCallback startTimeCal
 }
 
 func (s *Storage) Notify() {
+	s.mtx.Lock()
```

Comment:
`s.rws.mtx.Lock` is sufficient as `s.rws.ApplyConfig` called from `ApplyConfig` is the one that really conflicts with `Notify`, but I didn't see any use of inner locks like that.

Reply:
I think it's a better idea if it's locked inside `rws`. I don't see any other cases where inner fields of `rws` are touched by `Storage`.
So the loop for `Notify`, including lock, should be in a method on `WriteStorage`, and this method just forwards to that.